dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/commands/ProcessorUtils.java

```diff
@@ -261,7 +304,49 @@ private static File processArchive(String looseFileName, LooseConfig looseConfig
         } finally {
             Utils.tryToClose(thisArchive);
         }
+    }
 
+    /**
+     * Common code for generating the entryPath
+     *
+     * @param looseFile
+     * @param bootProps
+     * @param archiveEntryPrefix
+     * @param isUsr
+     * @return String
+     */
+    private static String generateBaseEntryPath(File looseFile, BootstrapConfig bootProps, String archiveEntryPrefix, boolean isUsr) {
+        File usrRoot = bootProps.getUserRoot();
+        int len = usrRoot.getAbsolutePath().length();
+
+        String entryPath = null;
+        if (archiveEntryPrefix.equalsIgnoreCase(PackageProcessor.PACKAGE_ARCHIVE_PREFIX) || !isUsr) {
+            entryPath = archiveEntryPrefix + BootstrapConstants.LOC_AREA_NAME_USR + looseFile.getAbsolutePath().substring(len);
+        } else {
+            entryPath = archiveEntryPrefix + looseFile.getAbsolutePath().substring(len);
+        }
+        entryPath = entryPath.replace("\\", "/");
+        entryPath = entryPath.replace("//", "/");
+        entryPath = entryPath.substring(0, entryPath.length() - 4); // trim the .xml
```

Comment:
I know this was copied from existing code - but its ugly - what is being trimmed 4 chars?

Reply:
```suggestion
        return entryPath.replace("\\", "/").replaceAll("/(?=/)|.xml$", "");
```
This is the cleanest way I could think of. It also handles any size group of slashes now instead of just pairs of two and trims the .xml explicitly.